app/src/rgb_underglow.c

```diff
@@ -369,15 +375,18 @@ struct zmk_led_hsb zmk_rgb_underglow_calc_sat(int direction) {
 
 struct zmk_led_hsb zmk_rgb_underglow_calc_brt(int direction) {
     struct zmk_led_hsb color = state.color;
-
-    int b = color.b + (direction * CONFIG_ZMK_RGB_UNDERGLOW_BRT_STEP);
-    if (b < 0) {
+    int b = color.b;
+    if (direction > 0 && b < CONFIG_ZMK_RGB_UNDERGLOW_BRT_MIN) {
+        b = CONFIG_ZMK_RGB_UNDERGLOW_BRT_MIN;
+    } else {
+        b += (direction * CONFIG_ZMK_RGB_UNDERGLOW_BRT_STEP);
+    }
+    if (b < CONFIG_ZMK_RGB_UNDERGLOW_BRT_MIN) {
         b = 0;
-    } else if (b > BRT_MAX) {
-        b = BRT_MAX;
+    } else if (b > CONFIG_ZMK_RGB_UNDERGLOW_BRT_MAX) {
+        b = CONFIG_ZMK_RGB_UNDERGLOW_BRT_MAX;
     }
     color.b = b;
-
```

Comment:
This logic seems odd to me. On 379, how would brightness be less than the minimum if we're already enforcing that and haven't adjusted it? Then we're also checking brightness min again on line 384. How about we do something like this?

```c
int b = color.b + (direction * CONFIG_ZMK_RGB_UNDERGLOW_BRT_STEP);

if (b < CONFIG_ZMK_RGB_UNDERGLOW_BRT_MIN) {
    b = CONFIG_ZMK_RGB_UNDERGLOW_BRT_MIN;
}

if (b > CONFIG_ZMK_RGB_UNDERGLOW_BRT_MAX) {
    b = CONFIG_ZMK_RGB_UNDERGLOW_BRT_MAX;
}

color.b = b;
```

With this, we step the brightness and then correct for any out-of-bounds brightnesses. I think this is a bit more straightforward, but I'm open to other ways of handling it too.